### DISCUSSION TITLE
Chamadas assíncronas em métodos assíncronos

### DISCUSSION THREAD

**Discussion by Arguto** [balta-io/forum#514] [Tira Dúvidas] [answered]
Boa tarde,

Estou dando manutenção em um projeto, onde preciso consumir uma API de terceiros. Porém, todos os métodos deste projeto são síncronos e quando faço chamadas assíncronas que precisei implementar, o método síncrono finaliza antes delas encerrarem. Por exemplo, preciso obter um token para autenticar em um api, porém, antes de eu receber o token, que é uma rotina assíncrona, a síncrona é encerrada. 
Eu não posso deixar todas as rotinas que já existem assíncronas, para usar o await. Porém, preciso aguardar que estas requisições encerrem para que o fluxo continue normalmente. Sou meio que iniciante em C# ainda, eu consigo consumir API's sincronamente? Ou há alguma outra forma de eu garantir que a requisição assíncrona conclua para que a síncrona continue?

Grato!

**Comment by PriciLeite** [comment, marked as answer]
GetTokenAsync().GetAwaiter().GetResult() para esperar pelo resultado de forma síncrona. Isso garante que o fluxo síncrono aguarde a conclusão da chamada assíncrona...
 
using System;
using System.Net.Http;
using System.Threading.Tasks;

class Program
{
    static void Main()
    {
        // Crie um método síncrono que chama um método assíncrono
        var token = GetTokenAsync().GetAwaiter().GetResult();

        // Continue com o fluxo síncrono
        Console.WriteLine($"Token recebido: {token}");
    }

    static async Task<string> GetTokenAsync()
    {
        using (var client = new HttpClient())
        {
            var response = await client.GetAsync("https://exemplo.com/api/token");
            response.EnsureSuccessStatusCode();
            return await response.Content.ReadAsStringAsync();
        }
    }
}

**Reply by Arguto** [reply]
Muito obrigado, era o que eu precisava. Mais um aprendizado!
